python/lsst/qserv/tests/benchmark.py

```diff
@@ -162,14 +162,22 @@ def runQueries(self, mode, dbName, stopAt=MAX_QUERY):
         _LOG.debug("Running queries : (stop-at: %s)", stopAt)
         if mode in ('qserv', 'qserv_async'):
             withQserv = True
-            sqlInterface = cmd.Cmd(config=self.config,
-                                   mode=const.MYSQL_PROXY,
-                                   database=dbName)
+            if (not qservServer):
+                sqlInterface = cmd.Cmd(config=self.config,
+                                       mode=const.MYSQL_PROXY,
+                                       database=dbName)
+            else:
+                conf = self.config
+                conf['qserv']['master'] = qservServer
+                _LOG.debug(" conf=%s", conf)
+                sqlInterface = cmd.Cmd(conf,
+                                       mode=const.MYSQL_PROXY,
+                                       database=None)
```

Comment:
There is an unneeded repetition in both `if` clauses, just do:

    if qservServer:
        self.config['qserv']['master'] = qservServer
    sqlInterface = cmd.Cmd(config=self.config,
                           mode=const.MYSQL_PROXY,
                           database=dbName)

(this assumes you are going to fix things so that default database name works 😉 )

Reply:
That is actually a really difficult problem to solve and I expect always getting schema, users, grants, etc. from a central location (or slaved replica thereof) to be far simpler than trying to keep several czars local databases synced.